### ISSUE TITLE
Multiple request calls are not working

### ISSUE THREAD

**Issue by axenteoctavian** [WalletConnect/WalletConnectSharp#77]
I'm just trying to `Request<T, TR>` multiple times. The first time is working, the second time I get this exception:
(First time I request with a RpcMethod and second time with different RpcMethod)
![image](https://user-images.githubusercontent.com/45915201/222415517-14bdc5d5-3c93-474e-aa0d-a318e7ab2c75.png)

I can't tell if this issue is on my side or from WC. Can you help me with an answer please?
Thank you!

**Comment by 0cv** [comment]
I had the same issue when I was trying to call multiple times `await WalletConnectSignClient.Init(dappOptions)`. This shall only be called in an init method, this was at least the issue for me

**Comment by axenteoctavian** [comment]
No, no. I call Init only once.
My issue is when using different RpcMethod attribute in Request

**Comment by tyronbrand** [comment]
@axenteoctavian did you ever resolve this?

**Comment by axenteoctavian** [comment]
@tyronbrand, I don't think there is a way to get over this unless you modify the WalletConnectSharp files.

Some suggested to modify theses [lines](https://github.com/WalletConnect/WalletConnectSharp/blob/d0a82f6263d26e0828d03511d8b4d47701f56713/WalletConnectSharp.Core/Controllers/JsonRpcHistory.cs#L35) with this code
```
public string Name
{
    get
    {
        if (typeof(T).IsGenericType)
        {
            return $"{_core.Name}-history-of-type-{typeof(T).GenericTypeArguments[0]}";
        }
        return $"{_core.Name}-history-of-type-{typeof(T).Name}";
    }
}
```

But I don't know if this is a proper change but it's working.
I'm waiting for an official answer but no response in 3 months unfortunately.

**Comment by tyronbrand** [comment]
I will give it a go, thanks for the fast response.

**Comment by Executus** [comment]
I had the same problem. The change suggested by @axenteoctavian worked for me.

**Comment by gigajuwels** [comment]
This issue has been addressed in 2.0.5 can you please confirm
